crates/bevy_text/src/text2d.rs

```diff
@@ -92,7 +92,12 @@ pub fn extract_text2d_sprite(
                 let transform = text_transform.mul_transform(glyph_transform);
 
                 extracted_sprites.sprites.push(ExtractedSprite {
-                    transform,
+                    transform: Extracted2dTransform {
+                        translation: transform.translation.truncate(),
+                        scale: transform.scale.truncate(),
+                        rotation: transform.rotation,
+                    },
+                    z_layer: transform.translation.z * transform.scale.z,
```

Comment:
```suggestion
                    z_layer: transform.translation.z,
```

The sprite has 0 width, changing the depth here based on the scale doesn't make sense.